### PR TITLE
create SnapshotWithBlobs interface 

### PR DESCRIPTION
**Pull request by dannimad**
-Unify snapshotree and its blobs types into a single interface SnapshotWithBlobs given that they refer to the same summary data and represent just different parts of the same summary.
-Move PendingContainerState interfaces to serializedStateManager
-Fix tests types

There are a couple of changes pending:
1. Get rid of ISnapshotTreeWithBlobContents which unfortunately it's harder to do than it seems.
2. Initialize this.snapshot with pendinglocalstate value from the constructor. There is no need to wait until fetchSnapshot.

### CODE REVIEW

File: packages/loader/container-loader/src/serializedStateManager.ts [hunk 1]
```diff
@@ -23,17 +23,53 @@ import {
...
+}
+/**
+ * State saved by a container at close time, to be used to load a new instance
+ * of the container to the same state
+ * @internal
+ */
+export interface IPendingContainerState {
+	attached: true;
+	pendingRuntimeState: unknown;
+	pendingSnapshot: PendingSnapshot;
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/serializedStateManager.ts:47]: these are serialized state, so we shouldn't change them

File: packages/loader/container-loader/src/serializedStateManager.ts [hunk 2]
```diff
@@ -23,17 +23,53 @@ import {
 	UsageError,
 	createChildMonitoringContext,
 } from "@fluidframework/telemetry-utils";
-import { IPendingContainerState } from "./container.js";
 import { ISerializableBlobContents, getBlobContentsFromTree } from "./containerStorageAdapter.js";
 
+export interface PendingSnapshot {
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/serializedStateManager.ts:28]: i don't thing we should use pending in the name here, as it is used in cases besides pending. maybe just call it SnapshotWithBlobs

File: packages/loader/container-loader/src/serializedStateManager.ts (hunk 1 above)
Comment (anthony-murphy) [reply packages/loader/container-loader/src/serializedStateManager.ts:47]: you could have this interface extend the above

File: packages/loader/container-loader/src/serializedStateManager.ts [hunk 3]
```diff
@@ -23,17 +23,55 @@ import {
...
+/**
+ * State saved by a container at close time, to be used to load a new instance
+ * of the container to the same state
+ * @internal
+ */
+export interface IPendingContainerState extends SnapshotWithBlobs {
+	attached: true;
+	pendingRuntimeState: unknown;
+	snapshotTree: ISnapshotTree;
+	snapshotBlobs: ISerializableBlobContents;
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/serializedStateManager.ts:48]:
```suggestion
```

File: packages/loader/container-loader/src/serializedStateManager.ts [hunk 4]
```diff
@@ -23,17 +23,55 @@ import {
...
 	createChildMonitoringContext,
 } from "@fluidframework/telemetry-utils";
-import { IPendingContainerState } from "./container.js";
 import { ISerializableBlobContents, getBlobContentsFromTree } from "./containerStorageAdapter.js";
 
+export interface SnapshotWithBlobs {
+	/**
+	 * Snapshot from which container initially loaded.
+	 */
+	snapshotTree: ISnapshotTree;
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/serializedStateManager.ts:32]:
```suggestion
	baseSnapshot: ISnapshotTree;
```
you can't change the property name for pending state

File: packages/loader/container-loader/src/serializedStateManager.ts [hunk 5]
```diff
@@ -23,17 +23,55 @@ import {
...
+
+/**
+ * State saved by a container in detached state, to be used to load a new instance
+ * of the container to the same state (rehydrate)
+ * @internal
+ */
+export interface IPendingDetachedContainerState extends SnapshotWithBlobs {
+	attached: false;
+	snapshotTree: ISnapshotTree;
+	snapshotBlobs: ISerializableBlobContents;
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/serializedStateManager.ts:67]:
```suggestion
```

File: packages/loader/container-loader/src/serializedStateManager.ts (hunk 4 above)
Comment (dannimad) [reply packages/loader/container-loader/src/serializedStateManager.ts:32]: what's the issue on doing that if we change the name on all places where it's being read?

Comment (dannimad) [reply packages/loader/container-loader/src/serializedStateManager.ts:32]: The interface is being used internally and the feature is not public so there shouldn't be any problems for backward compatibility.

Comment (anthony-murphy) [reply packages/loader/container-loader/src/serializedStateManager.ts:32]: any exist usage of the serialized data will break, like how this is breaking the snapshot tests

**Review by anthony-murphy — APPROVED** [review]

**Pull request merged**